### PR TITLE
Error on defined but empty DOCKER_IMAGE

### PR DESCRIPTION
**Pull request by machinekoder**
Check if DOCKER_IMAGE environment variable was set.
If it is set and empty, throw an error.

See https://github.com/ros-industrial/industrial_ci/issues/296
Supersedes #297

### CODE REVIEW

**Comment by machinekoder** [comment]
@ipa-mdl What's the best way to test this? I don't understand how `source_tests.sh` works.

**Comment by ipa-mdl** [comment]
I have just added the test :)

**Comment by ipa-mdl** [comment]
The tests passed, but only the first line is colored. I would keep it this way.

File: industrial_ci/src/docker.sh [hunk 1]
```diff
@@ -164,8 +164,10 @@ function ici_prepare_docker_image() {
     else # url, run directly
         ici_docker_build "$DOCKER_FILE" > /dev/null
     fi
-  elif [ -z "$DOCKER_IMAGE" ]; then # image was not provided, use default
+  elif [ -z ${DOCKER_IMAGE+x} ]; then # image was not provided, use default
```
Comment (ipa-mdl) [inline industrial_ci/src/docker.sh]: Why did you remove the quotes?

**Comment by 130s** [comment]
I took advantage of maintainership and pushed a few commits to @machinekoder's branch to address some comments.

**Comment by ipa-mdl** [comment]
I think it is ready for a squash(!)-merge, but I am concerned that this might break some test set-ups.
There is no way to unset a global variable in Travis :-/
Until now it was sufficient to just empty the variable for specific jobs.

**Comment by ipa-mdl** [comment]
supersed by #617
